packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -269,27 +269,31 @@ export default util.createRule<Options, MessageId>({
     function checkProperty(node: TSESTree.Property): void {
       const tsNode = parserServices.esTreeNodeToTSNodeMap.get(node);
       if (ts.isPropertyAssignment(tsNode)) {
+        if (!returnsThenable(checker, tsNode.initializer)) {
+          return;
+        }
         const contextualType = checker.getContextualType(tsNode.initializer);
         if (
           contextualType !== undefined &&
           isVoidReturningFunctionType(
             checker,
             tsNode.initializer,
             contextualType,
-          ) &&
-          returnsThenable(checker, tsNode.initializer)
+          )
         ) {
           context.report({
             messageId: 'voidReturnProperty',
             node: node.value,
           });
         }
       } else if (ts.isShorthandPropertyAssignment(tsNode)) {
+        if (!returnsThenable(checker, tsNode.name)) {
+          return;
```

Comment:
> Added line #L291 was not covered by tests

If these lines aren't getting hit by tests, then one of two things is possible:

* These lines are unhittable and can be removed
* The tests are missing cases that your code does hit

Do you have examples of the latter? If not, I'd assume it's the former and we can remove the lines.